Support arbitrary prefill lengths in Mixtral demo

sraizada-tt opened the pull request:
### Ticket
#10934 

### Problem description
Support arbitrary prefill lens in Mixtral demo

### What's changed
* Support arbitrary prefill lens in Mixtral demo
* Remove any decode of input prompt 
* Support different pos_ids for each user 

### Checklist
- [ ] Post commit CI passes
- [ ] T3k Demo
- [ ] T3k frequent
- [ ] T3k model perf

mtairum committed:
#10934: Initial commit for 32k Mixtral demo

mtairum committed:
#10934: Added support for 32k Mixtral demo

mtairum committed:
#10934: Fixes to Mixtral demo with prefill. Added support to >16k prompts.

mtairum committed:
#10934: Add support for instruct weight

sraizada-tt committed:
#5337: rot mat fix

mtairum committed:
#10934: Revamped the entire Mixtral demo with large input

mtairum committed:
#0: [REMOVE COMMIT] Disable all other t3k CI demo tests

mtairum committed:
#10934: Changed Mixtral demo in CI to run instruct weights

mtairum committed:
#10934: Changed location of 32k input for Mixtral demo in CI env

mtairum committed:
#10934: Minor fix

mtairum committed:
#10934: Add 16k input to mixtral

mtairum committed:
#10934: Updated Mixtral README

mtairum committed:
#10934: Updated timeout of Mixtral demo and PCC for mixtral model test

mtairum committed:
#10934: Removed token matching validation due to non-deterministic outputs

mtairum committed:
Revert "#0: [REMOVE COMMIT] Disable all other t3k CI demo tests"

This reverts commit 7fbfa405369cae2bb165f7b3f604a2dc54992763.

mtairum committed:
#10934: Incresaed the t3k mixtral demo test timeout

sraizada-tt committed:
#5337: make start_pos a list

sraizada-tt committed:
#5337: demo ouputs look good

sraizada-tt committed:
#5337: Merge branch 'main' of https://github.com/tenstorrent/tt-metal into mixtral-32k-prefill

mtairum committed:
#10934: fix merge regression

mtairum commented on models/demos/t3000/mixtral8x7b/tt/mixtral_common.py:
```diff
@@ -111,102 +116,70 @@ def preprocess_inputs_prefill(
+    #     for i in range(max_prompt_len - prefill_seq_len)
+    # ]
+
+    input_tokens_prefill_tt = None
```
Are these safe to be removed? From here and from everywhere where `preprocess_inputs_prefill()` is called?

mtairum committed:
#10934: Small bugfixes and code cleanup

mtairum committed:
#10934: Update Mixtral CI test to the 32k prefill case

sraizada-tt committed:
#5337: bug fixes

mtairum committed:
#10934: Merge branch 'main' into mixtral-32k-prefill

> Conflicts:
>	models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf_benchmark.py

mtairum reviewed (approved):
I'm re-running all tests locally and on CI before merging to main. 

There were some minor issues with recent merge changes (in the future let's aim for frequent rebases instead), so want to play it safe.

mtairum committed:
#10934: Re-enable output token matching in Mixtral

mtairum committed:
#10934: Also re-enable output token matching in llama3.1-8b

sraizada-tt committed:
Merge branch 'main' into mixtral-32k-prefill

mtairum committed:
#10934: [Mixtral] Update default max_seq_len to 32k

sraizada-tt committed:
#5337: good instruct demo outputs

mtairum committed:
#5337: Merge branch 'main' into mixtral-32k-prefill

mtairum committed:
#5337: Remove cutoff for 128seqlen

mtairum committed:
#5337: Merge branch 'main' into mixtral-32k-prefill

> Conflicts:
>	models/demos/t3000/mixtral8x7b/demo/demo.py
>	models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_attention.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_decoder.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_model.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_model_prefill.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf_benchmark.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_perplexity.py
>	models/demos/t3000/mixtral8x7b/tests/test_mixtral_topk.py
>	models/demos/t3000/mixtral8x7b/tt/mixtral_attention.py
>	models/demos/t3000/mixtral8x7b/tt/mixtral_common.py
>	models/demos/t3000/mixtral8x7b/tt/mixtral_model.py

mtairum committed:
#5337: Fix some merge conflict errors

mtairum committed:
#5337: Fix yet another merge conflict

mtairum committed:
#5337: Merge branch 'main' into mixtral-32k-prefill

mtairum committed:
#0: [REVERT COMMIT] Change Mixtral output print for debugging

mtairum committed:
#5337: Fix merge regression

mtairum committed:
#5337: Merge branch 'main' into mixtral-32k-prefill

mtairum committed:
#5337: Remove expected outputs

mtairum committed:
Merge branch 'main' into mixtral-32k-prefill

mtairum committed:
Merge branch 'main' into mixtral-32k-prefill

mtairum commented:
Merging this PR has it provides the backbone for a couple more changes currently in flight, and thus will enable some work parallelization without the hurdles of merge conflicts and such.

There is currently a P0 investigation brought forward by the changes in this PR here: https://github.com/tenstorrent/tt-metal/issues/12206
